### PR TITLE
Makes Save As Draft non-destructive

### LINKED ISSUE #6761: Unexpected "Save as Draft" appearance/behavior
This was originally reported by @rachelmcr [here](https://github.com/wordpress-mobile/WordPress-Aztec-iOS/issues/296).  I moved it here for organizational purposes.

### Expected behavior

In the visual editor, the ellipsis menu (top right) action sheet has all blue text, including "Save as Draft." When you select that option, the editor stays open so you can continue editing:

### Actual behavior 

In Aztec, the ellipsis menu's “Save as Draft” option is marked in red. When you select that option, the editor closes unexpectedly:

@astralbodies noted that the red text was intentional:

>  I made Save as Draft red because it has a behavior that fits Apple’s guidelines for it to be called out like that.

If that color change better fits the design guidelines that makes sense to me, but I'd expect the behavior to stay the same (keeping the editor open).

##### Tested on iPhone 6, iOS 10.2.1, WPiOS 7.0.0.20170214

### PR DESCRIPTION
**Pull request by astralbodies**
Fixes #6761  

This makes the Save as Draft button in the options menu non-destructive (blue instead of red). It also fixed everything else that went wrong when no longer dismissing the window when it's tapped. 

* Now when the post is published/pushed to the server the post is replaced with a new revision if the window isn't immediately dismissed (matches original hybrid VC logic). 
* Made window dismissal logic more explicit instead of saying the secondary publish button was tapped. Code is much clearer to follow.

## To test:
1. Create a new post.
2. Tap the options menu. 
3. Verify Save as Draft is present and in non-destructive colors.
4. Tap Save as Draft.
5. Verify Publish button changes to Update and the window is not dismissed. Note: "dirty" logic isn't available yet so the publish button will continue to be enabled instead of disabling until more changes are made.

Next:

1. Edit an existing draft.
2. Tap the options menu.
3. Verify Publish Immediately is present and is in destructive colors (red).
4. Tap Publish Immediately.
5. Verify window dismisses after publishing is complete. (post-post will appear)

Needs review: @diegoreymendez, @rachelmcr (if you have time to install the alpha from Buddybuild)

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Post/AztecPostViewController.swift [hunk 1]
```diff
@@ -569,18 +570,18 @@ extension AztecPostViewController {
 //
 extension AztecPostViewController {
     @IBAction func publishButtonTapped(sender: UIBarButtonItem) {
-        handlePublishButtonTapped(secondaryPublishTapped: false)
+        handlePublishButtonTapped(shouldDismissEditorWhenDone: true)
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/Post/AztecPostViewController.swift]: IMO `shouldDismissEditorWhenDone` is a bit too wordy for swift (so is `handlePublishButtonTapped` 😝). How about something like `publishTapped(dismissWhenDone:)` or `publishTapped(autoDismiss:)`?

Comment (astralbodies) [inline WordPress/Classes/ViewRelated/Post/AztecPostViewController.swift]: Agreed - making those changes right now.

**Comment by rachelmcr** [comment]
I ran into a small issue saving a new post:

> 5. Verify Publish button changes to Update and the window is not dismissed.

This isn't working as expected for me. After tapping Save as Draft, the draft is saved but the Publish button changes to Save instead of Update. (If I close the editor and reopen it, the button is Update as expected.)

Publishing an existing draft works, except for this wording:

> 3. Verify Publish Immediately is present and is in destructive colors (red).

I see the text Publish Now rather than Publish Immediately.

(Tested using build 2661 from Buddybuild.)

**Comment by astralbodies** [comment]
Nice catch @rachelmcr - not sure how I missed that!!

> but the Publish button changes to Save instead of Update.

Turns out the state is actually changing but the button title wasn't being updated. I updated the code so that the button title is refreshed in all cases.

> I see the text Publish Now rather than Publish Immediately.

I meant Publish Now - oops 😄

**Comment by astralbodies** [comment]
@rachelmcr Do you have time to give this one another look?

**Comment by jleandroperez** [comment]
One minor note: **Save as Draft** is available, even when the post has no title / content.
Pressing it results in an error onscreen.

**Comment by diegoreymendez** [comment]
I keep getting the impression that the editor should just "Save", and that publishing should be a separate UI/UX triggered from the list of posts.

**Comment by astralbodies** [comment]
@rachelmcr:

> If I create a new post and add a title but no post content, or post content but no title, the "Save as Draft" option does not appear. I should be able to save a draft with either a title or post content (or both).

There was a bug in the PR #6818 which I just resolved in 938dcf6 that fixes this issue. Post title OR post body having content will enable the publish button in the nav bar and make save draft show up in the options menu.

> If I open an existing draft, the "Publish Now" option does not appear right away.

This should also be resolved.

**Comment by rachelmcr** [comment]
Looks good to me now! :)

**Comment by astralbodies** [comment]
Thanks for all the reviews!
